R/pkg/R/utils.R

```diff
@@ -334,6 +334,27 @@ varargsToEnv <- function(...) {
   env
 }
 
+# Utility function to capture the varargs into environment object but all values are converted
+# into string.
+varargsToStrEnv <- function(...) {
+  pairs <- list(...)
+  env <- new.env()
+  for (name in names(pairs)) {
+    value <- pairs[[name]]
+    if (!(is.logical(value) || is.numeric(value) || is.character(value) || is.null(value))) {
+      stop("value[", value, "] in key[", name, "] is not convertable to string.")
```

Comment:
this might not be ideal because the user is not calling this function directly and `value[something]` might not mean anything to them (since they have never set any value thing, furthermore, that might not be the relevant syntax in R)

Any idea on a different way to report this?